LiteFarmOrg/LiteFarm
# Lf 2534 refactor sensors to be locations

Ticket: [LF-2534](https://lite-farm.atlassian.net/browse/LF-2534)
Related Ticket: [LF-2510](https://lite-farm.atlassian.net/browse/LF-2510)

Refactored backend and frontend logic to include sensors as a location. 

To test:
- Run the latest migrations
  - You may need to remove the grid points column with the following sql: `ALTER TABLE sensor DROP COLUMN grid_points;`
- Make sure you have the following environment variables in `packages/api/.env`:
```
ENSEMBLE_USERNAME=?
ENSEMBLE_PASSWORD=?
```  
- Go to the farm map -> "+" -> scroll down and click sensor
- Use a csv file with a valid format and with valid esids, and upload that file (ex: [testBulkUpload.csv](https://github.com/LiteFarmOrg/LiteFarm/files/8951095/testBulkUpload.csv) )

- You should see a success banner, and the page will refresh with the sensors you inputted on the map. 
  - You may need to unclaim the sensor if you want to retest
- Also check if the sensor map filter is working. If not this PR can still move forward.
